fg/settings/main.py

```diff
@@ -112,4 +112,7 @@
 try:
     from .secrets import *
 except ImportError:
-    pass
+    try:
+        from .env_secrets import *
```

Comment:
Why do we need both?

Reply:
I push to my Dokku / Heroku servers the entire git without any modifications (no .gitignore'd files get sent). If the system doesn't see any secrets.py, it assumes that we want to use environmental variables instead.

Reply:
Ah, so dokku let's you define a custom environment somewhere else? Can you walk the user (err me) through this in the docs? And also it would be good to put a comment above that, something like:

```python
# For deployments that rely entirely on the environment, a secrets.py won't exist, but variables will be read from `env_secrets.py`
```
And in this case, the user will also need to know that he/she is in charge of editing the env_secrets.py for authentication (and other secrets that may/may not be defined) so please put a note about this under Settings / Secrets in the setup.md too.

Reply:
Actually, I'd really like to test this out with how you do it dokku (maybe this weekend?) so write the `docs/_docs/development/deploy/dokku.md` like you are talking to someone that has absolutely no idea what they are doing. (ahem, me!) And after we merge the PR here, I'll re-organize the docs a bit so that Google Cloud is organized under that new deploy folder as another option.

Reply:
`like you are talking to someone that has absolutely no idea what they are doing. (ahem, me!) And after we merge the PR here, I'll re-organize the docs a bit so that Google Cloud is organized under that new deploy folder as another option.`

Will do! Lemme get it working, and then I'll add the code + docs in and request another review.

`Ah, so dokku let's you define a custom environment somewhere else? `
Yep! The annoying part is you have to ssh into the Dokku server, and then define which environmental variables get imported. Once you do that, though, everything works very smoothly, and you can just `git push stage master` or `git push production master` and the code will be deployed to whichever server you want. Usually I keep a stage and a production, though I'll probably just set up another for personal development.

Reply:
Love it! And if there is some terrible issue that needs debugging, I'm guessing you can ssh in too (and the code that you pushed to git is there?). The stage / production server is useful for sure.

So I'm anticipating cloning the repo locally, adding my preferences for authentication, etc. in env_secrets.py, ssh-ing into the server to add some bash script (or other) to be for the environment (maybe dokku has a special named / located file for that) and then pushing to a branch for stage /production to deploy (and possibly having told the service to map those branches from that repo). 

Anyway don't spend any more time re-explaining things that you are going to write out in detail - looking forward to reading it!

Reply:
A little more complicated but also a little easier. You can ssh into the containers, but usually I just check the logs, modify code locally, and redeploy. When you're messing with the system itself, this gets a bit annoying, since you tend to commit a lot.

You would ssh into the dokku server and then run the command `dokku apps:create django-is-cool`, and then configure your environmental variables with `dokku config:set django-is-cool ADMIN_NAME=me`. Then, you just push the code to it.

One thing I like is how it doesn't really use docker-compose, you have to define those elsewhere, usually with plugins. For example, if I want a postgres database, I just run `dokku postgres:create django-db` and then `dokku postgres:link django-db django-is-cool` and boom! database is there.

Then, I can just backup the database with `dokku postgres:backup`, and set schedules with it.

Anyway, will write docs, almost at deployment I think, just one blocking problem.